chia/wallet/wallet_state_manager.py

```diff
@@ -641,68 +644,85 @@ async def unconfirmed_removals_for_wallet(self, wallet_id: int) -> Dict[bytes32,
 
     async def determine_coin_type(
         self, peer: WSChiaConnection, coin_state: CoinState, fork_height: Optional[uint32]
-    ) -> Optional[WalletIdentifier]:
+    ) -> Tuple[Optional[WalletIdentifier], Optional[VersionedBlob]]:
```

Comment:
Should we be using `VersionedBlob`?  It's not like this is going into a DB or anything, and you're forcing wallets to deserialize this when they get passed.  It might be better to define a `Protocol` or something for all of the `...CoinData` types and just return the object and have the wallet method assert that it's the right type of object.

Reply:
These types of data are quite different and don't have any common method, so I'm not sure a Protocol is necessary. Actually, I'm planning to save them in the database since we added a new column in the coin_store for metadata. It can help us prevent creating more coin-level tables, display history,  and debugging. But obviously, it will increase the db size.

Reply:
Right, maybe not a `Protocol` just like a superclass for all of them.  Serializing it here, then immediately de serializing it in the `coin_added` methods seems wasteful, if you want to store them in the DB, there's no harm in serializing them before you do that if/when you decide to.